tests/test_runtime.py

```diff
@@ -181,3 +183,87 @@ async def has_log() -> bool:
+    # Set up a Prometheus configuration with custom histogram bucket overrides
+    prom_addr = f"127.0.0.1:{find_free_port()}"
+    special_value = float(1234.5678)
+    histogram_overrides = {
+        "temporal_long_request_latency": [special_value / 2, special_value],
+        "custom_histogram": [special_value / 2, special_value],
+        "temporal_workflow_endtoend_latency": [
+            special_value / 2,
+            special_value,
+        ],
+    }
+
+    runtime = Runtime(
+        telemetry=TelemetryConfig(
+            metrics=PrometheusConfig(
+                bind_address=prom_addr,
+                counters_total_suffix=False,
+                unit_suffix=False,
+                durations_as_seconds=False,
+                histogram_bucket_overrides=histogram_overrides,
+            ),
+        ),
+    )
+
+    # Create a custom histogram metric
+    custom_histogram = runtime.metric_meter.create_histogram(
+        "custom_histogram", "Custom histogram", "ms"
+    )
+
+    # Record a value to the custom histogram
+    custom_histogram.record(600)
+
+    # Create client with overrides
+    client_with_overrides = await Client.connect(
+        client.service_client.config.target_host,
+        namespace=client.namespace,
+        runtime=runtime,
+    )
+
+    async def run_workflow(client: Client):
+        task_queue = f"task-queue-{uuid.uuid4()}"
+        async with Worker(
+            client,
+            task_queue=task_queue,
+            workflows=[HelloWorkflow],
+        ):
+            assert "Hello, World!" == await client.execute_workflow(
+                HelloWorkflow.run,
+                "World",
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=task_queue,
+            )
+
+    await run_workflow(client_with_overrides)
+
+    async def check_metrics() -> bool:
+        try:
+            with urlopen(url=f"http://{prom_addr}/metrics") as f:
+                metrics_output = f.read().decode("utf-8")
+
+                for key, buckets in histogram_overrides.items():
+                    if key not in metrics_output:
+                        return False
+                    for bucket in buckets:
+                        # expect to have {key}_bucket and le={bucket} in the same line with arbitrary strings between them
+                        regex = re.compile(f'{key}_bucket.*le="{bucket}"')
+                        if regex.search(metrics_output) is None:
+                            return False
+                return True
+        except Exception as e:
+            # Log the error for debugging if needed, but return False so assert_eventually retries
+            print(f"Error fetching/checking metrics: {e}")
+            return False
+
+    # Wait for metrics to appear and match the expected buckets
+    await assert_eq_eventually(
+        True,
+        check_metrics,
+        timeout=timedelta(seconds=30),
+        interval=timedelta(seconds=2),
```

Comment:
```suggestion
```

I think the defaults are better here. We don't want to sleep for two seconds or wait 30, I think 10s at 200ms interval is reasonable.